### PR TITLE
Added gemmlowp package

### PR DESCRIPTION
**Pull request by Muakasan**

### CODE REVIEW

File: var/spack/repos/builtin/packages/gemmlowp/package.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Gemmlowp(Package):
+    """Google low-precision matrix multiplication library"""
+
+    homepage = "https://github.com/google/gemmlowp"
+    version('a6f29d9ac', git='https://github.com/google/gemmlowp.git',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gemmlowp/package.py]: The problem with calling it develop is, what happens when someone adds a newer commit? You can't also call that develop.

I think we should ask or encourage the developers to make releases. But obviously we have no control over that. For packages that don't make releases and expect all users to use the latest version, they often don't support problems in older commits. We should honor there wishes and simply install the latest develop or master branch, whichever they recommend.

I would also like to hear @citibeth's thoughts. I know there are security concerns about this approach.

Comment (davydden) [inline var/spack/repos/builtin/packages/gemmlowp/package.py]: I would remove a specific commit here. But I fully agree, there should be some releases, even 0.01

**Comment by citibeth** [comment]
Can we keep it please?  This issue has already been discussed.

**Comment by tgamblin** [comment]
Hey guys.  For the hackathon here, we used specific commit ids because tensorflow wants specific commit ids.  It's likely that if we incorporate tensorflow into spack that it will continue to want specific commit ids, and it doesn't seem like Google provides (or will provide soon) versions or releases for these libraries.

So in this case the intent is actually not to make this into a "latest release" branch, like we usually use `develop`.  It's really to refer to a specific commit that tensorflow wants.  So I think the commit id is more descriptive.  We could think about adding a date or maybe using the TF version that wants this commit, but I'm not sure if it's worth coming up with an identifier scheme when the main dependent refers directly to the commit, anyway.

File: var/spack/repos/builtin/packages/gemmlowp/package.py (hunk 1 above)
Comment (tgamblin) [inline var/spack/repos/builtin/packages/gemmlowp/package.py]: @davydden: we're likely to need more references to specific commits as tensorflow evolves.  We only packaged the latest version, so there is only one version here now, but this isn't intended to be a reference to the top of `develop`. It really is a specific reference.

Comment (tgamblin) [inline var/spack/repos/builtin/packages/gemmlowp/package.py]: Obviously though if the google folks ever add versions we'd use them...

**Comment by davydden** [comment]
@citibeth good idea

**Comment by adamjstewart** [comment]
I'm fine with using git commits as versions if that's how tensorflow works. We lose the ability to sort by version, and we'll need to use `preferred=True`, but it's better than making our own fake releases. What happens when someone contacts the developer and complains that they can't get `0.0.0.1` to install? The developer would be pretty confused. But if we use the commit hash, it's more descriptive.

**Comment by citibeth** [comment]
We've used "fake" releases elsewhere.  It's not just sorting by version we
lose the ability to do; we lose the ability to control anything by
version.  Which is used extensively in our packages.

**Pull request merged**